tests/ImageSharp.Tests/Common/SimdUtilsTests.cs

```diff
@@ -105,29 +105,13 @@ public void FastRound_RandomValues(int seed, float scale)
             AssertEvenRoundIsCorrect(r, v);
         }
 
-        private bool SkipOnNonAvx2([CallerMemberName] string testCaseName = null)
-        {
-            if (!SimdUtils.HasVector8)
-            {
-                this.Output.WriteLine("Skipping AVX2 specific test case: " + testCaseName);
-                return true;
-            }
-
-            return false;
-        }
-
-        [Theory]
+        [IntrinsicTheory(_HwIntrinsics.AVX2)]
```

Comment:
This change will skip some tests which should not be skipped.

The tests and the code under test are here from the times HW Intrinsics did not exist. `SkipOnNonAvx2` isn't really checking for AVX2, but rather for `Vector.IsHardwareAccelerated && Vector<float>.Count == 8`, which is poor man's AVX2 on older platforms.

When we started to add `System.Runtime.Intrinsics.X86` stuff to the repo, I updated the product code to use the term *"Vector8"* instead of *"Avx2"* for the old `Vector<float>` stuff, but looks like the test code still keeps using Avx2, which is confusing.

Reply:
Hm, makes sense now, I will add `Vector8` value separate from avx2 and check it with `Vector.IsHardwareAccelerated && Vector<float>.Count == 8`. Will also try to figure out which tests need that specific check. As this can be crucial to some tests will not unmark draft thingy. Thanks for clarifying, this one was the most confusing bit about intrinsics tests.

Reply:
A message so I would understand this properly, `Vector8` basically translates to avx intructions on runtimes without `Intrinsic` API platforms?

What is also a bit confusing, isn't first avx version also have vector8 support (256bit vectors to be hardware-friendly)? Wouldn't `Vector.IsHardwareAccelerated && Vector<float>.Count == 8` be true for avx but **not** avx2 CPUs?

And the most important question: should we care? 😂
Maybe checks should be like something like this:
// not sure if any other simd extension provides 256bit vectors
Avx -> `Vector.IsHardwareAccelerated && Vector<float>.Count == 8`
Avx2 -> `Avx2.IsSupported`

Reply:
And yet another message, *maybe* we should rename `SimdUtils.HasVector8` to `SimdUtils.HasAvx`?
it would comply with `HasAvx2` property:
```csharp
SimdUtils.HasVector8
SimdUtils.HasAvx2
```
vs
```csharp
SimdUtils.HasAvx  // either Vector8 check OR Avx.IsSupported
SimdUtils.HasAvx2 // Avx2.IsSupported
```

Reply:
The `Vector<float>.Count == 8` optimization actually uses AVX2, so `HasAvx` would be misleading, I prefer `Vector8` since it makes it clear we refer to the `Vector<float>` stuff.

Reply:
Oh now I got it, condition `Vector<float>.Count == 8` could mean Avx OR Avx2 and would compile operations to intrinsics if it's supported. So I guess it's better to provide all 3 variants: `Avx, Avx2, Vector8` so Avx2 specific code tests can use this.

Reply:
Not sure if you got this, but the way Vector<T> is implemented in the JIT, it sets the vector size to 128 bit unless AVX2 is supported.  So even though all float operations could be done at 256-bit width on processors supporting AVX and not AVX2, those processors still get `Vector<float>.Count == 4`.  `Vector<float>.Count == 8` really does imply AVX2 support.

Reply:
@saucecontrol that's super weird, thanks for clarifying. I've decided to implement Vector8 and avx/avx2 flags separately.

Reply:
The short explanation is that they map `Vector<T>` as a whole to either `TYP_SIMD16` or `TYP_SIMD32` in the JIT.  It can only be mapped to `TYP_SIMD32` if all operations on all `T` are supported, so that means AVX2.

The JIT code is really fascinating if you ever get a chance to dig into it.  The code that sets `Vector<T>` size is here, for example: https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/ee_il_dll.cpp#L307-L345

Reply:
Had yet another question about another `Vector<T>` detail but now I can have a truly full picture myself, super interesting. Thanks a lot!